src/Application/Commands/PatchFunds8Oct.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace MatchBot\Application\Commands;
+
+use Doctrine\ORM\EntityManagerInterface;
+use MatchBot\Application\Matching\Allocator;
+use MatchBot\Application\Matching\MatchFundsRedistributor;
+use MatchBot\Application\Messenger\DonationUpserted;
+use MatchBot\Domain\Campaign;
+use MatchBot\Domain\CampaignFundingRepository;
+use MatchBot\Domain\CampaignRepository;
+use MatchBot\Domain\Donation;
+use MatchBot\Domain\DonationRepository;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Messenger\RoutableMessageBus;
+
+/**
+ * @see MatchFundsRedistributor which has some of the core logic in common.
+ */
+#[AsCommand(
+    name: 'matchbot:patch-funds-8-oct',
+    description: 'Fix some incorrect fundings',
+)]
+class PatchFunds8Oct extends Command
+{
+    public function __construct(
+        private Allocator $allocator,
+        private CampaignRepository $campaignRepository,
+        private CampaignFundingRepository $campaignFundingRepository,
+        private DonationRepository $donationRepository,
+        private EntityManagerInterface $entityManager,
+        private LoggerInterface $logger,
+        private RoutableMessageBus $bus,
+    ) {
+        parent::__construct();
+    }
+
+    #[\Override]
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $campaigns = $this->getCampaignsAffected();
+        $output->writeln('Campaigns affected: ' . count($campaigns));
+
+        foreach ($campaigns as $campaign) {
+            $campaignId = $campaign->getId() ?? 0;
```

Comment:
the 0 isn't meaningful is it?
```suggestion
            $campaignId = $campaign->getId() ?? throw new \Exception('campaign id missing');
```